Snowflake.Data/Core/QueryContextCache.cs

```diff
@@ -0,0 +1,357 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using System.Linq;
+using Snowflake.Data.Log;
+
+namespace Snowflake.Data.Core
+{
+    internal class QueryContextValue
+    {
+        // base64 encoded string of Opaque information
+        [JsonProperty(PropertyName = "base64Data")]
+        public string Base64Data { get; set; }
+
+        public QueryContextValue(string context)
+        {
+            Base64Data = context;
+        }
+    }
+    internal class QueryContextValueEmpty
+    {
+        // empty object with no filed
+    }
+
+    internal class QueryContextElement
+    {
+        // database id as key. (bigint)
+        public long Id { get; set; }
+
+        // When the query context read (bigint). Compare for same id.
+        public long ReadTimestamp { get; set; }
+
+        // Priority of the query context (bigint). Compare for different ids.
+        public long Priority { get; set; }
+
+        // Opaque information (object with a value of base64 encoded string).
+        public string Context { get; set; }
+
+        public QueryContextElement(long in_id, long in_timestamp,
```

Comment:
Could we go with the camel case for the parameters.

Reply:
done